keras/backend/tensorflow_backend.py

```diff
@@ -1291,6 +1293,19 @@ def switch(condition, then_expression, else_expression):
     return x
 
 
+def select(condition, then_expression, else_expression):
+    """TensorFlow API: tf.select(condition, t, e, name=None)
+
+    `condition`, `then_expression` & `else_expression`
+    must all 3 be tensors of the same shape
+    """
+    # tf.expand_dims(condition, 0)
+    condition = tf.cast(tf.ones_like(then_expression) * tf.to_float(condition), dtype=bool)
+    condition.set_shape(copy.copy(then_expression.get_shape()))
+    # condition = tf.tile(condition, tf.pack([1, tf.shape(then_expression)[1]]))
+    return tf.select(condition, then_expression, else_expression)
```

Comment:
@uglyboxer: when I fit to a ParametricSoftExponential() layer/activation, this fills tensor output with NaNs